src/Sarif/Core/StackFrame.cs

```diff
@@ -0,0 +1,160 @@
+            int nativeOffset = dotNetStackFrame.GetNativeOffset();
+            MethodBase methodBase = dotNetStackFrame.GetMethod();
+            Assembly assembly = methodBase?.DeclaringType.Assembly;
+            string fullyQualifiedName = CreateFullyQualifiedName(methodBase);
+
+            PhysicalLocation physicalLocation;
+            
+            if (fileName != null)
+            {
+                physicalLocation = new PhysicalLocation()
+                {
+                    Uri = new Uri(fileName),
+                    Region = new Region()
+                    {
+                        StartLine = dotNetStackFrame.GetFileLineNumber(),
+                        StartColumn = dotNetStackFrame.GetFileColumnNumber(),
+                        Offset = ilOffset
+                    }
+                };
+            }
+            else
+            {
+                physicalLocation = new PhysicalLocation()
+                {
+                    Region = new Region() { Offset = ilOffset }
+                };
+            }
+
+            StackFrame stackFrame = new StackFrame()
+            {
+                FullyQualifiedLogicalName = fullyQualifiedName,
+                Location = physicalLocation,
+                Module = assembly?.GetName().Name,                 
+            };
+
+            if (nativeOffset != -1)
+            {
+                stackFrame.Properties = new Dictionary<string, string>
+                {
+                    { "NativeOffset", nativeOffset.ToString(CultureInfo.InvariantCulture) }
+                };
+            }
+
+            return stackFrame;
+        }
+
+        public override string ToString()
+        {
+            string result = AT + this.FullyQualifiedLogicalName;
+
+            if (Location?.Uri != null)
+            {
+                string lineNumber = Location.Region.StartLine.ToString(CultureInfo.InvariantCulture);
+                string fileName = Location.Uri.LocalPath;
+
+                result += IN + fileName + LINE + " " + lineNumber;
+            }
+
+            return result;
+        }
+
+        public static string CreateFullyQualifiedName(MethodBase methodBase)
+        {
+            if (methodBase == null) { return null;  }
+
+            var sb = new StringBuilder();
+
+            Type t = methodBase.DeclaringType;
+            // if there is a type (non global method) print it
+            if (t != null)
+            {
+                sb.Append(t.FullName.Replace('+', '.'));
+                sb.Append(".");
+            }
+            sb.Append(methodBase.Name);
+
+            // deal with the generic portion of the method
+            if (methodBase is MethodInfo && ((MethodInfo)methodBase).IsGenericMethod)
+            {
+                Type[] typars = ((MethodInfo)methodBase).GetGenericArguments();
+                sb.Append("[");
+                int k = 0;
+                bool fFirstTyParam = true;
```

Comment:
(although in this case the name would be something like `isFirstTypeParameter`. Still works for me.)